### PR TITLE
Adds ownership type to EntitlementInfo

### PR DESCRIPTION
**Pull request by daentech**
Description:

This PR adds `ownershipType` to the Android Purchases SDK

I know that the play store does not provide this information, but it is possible that a user id exists in multiple platforms and as such may have their entitlement purchased by themselves, or shared to them, for example on iOS, and then be using it on Android.

This information is stored in the revenucat API and as such, the entitlement could be returned to Android with this data and a store type of e.g. `APP_STORE`

I'm making use of this in the purchases-flutter SDK, so further PRs will be forthcoming to update both `purchases-hybrid-common` and `purchases-flutter`

  - [x] The issue number(s) or PR number(s) in the description if you are contributing in response to those.

Related to Zendesk ticket: 12668

  - [x] If applicable, unit tests.

### CODE REVIEW

**Comment by daentech** [comment]
Looks like the detekt is failing on stuff that I haven't changed (well, I added a new parameter to the constructor, but it was already too long):
```
TooManyFunctions - 11/11 - [Factories.kt] at /home/circleci/purchases-android/common/src/main/java/com/revenuecat/purchases/common/Factories.kt:1:1
	LongParameterList - 13/7 - [EntitlementInfo] at /home/circleci/purchases-android/public/src/main/java/com/revenuecat/purchases/EntitlementInfo.kt:29:34
```

And the assemble sample app is failing due to a missing/broken certificate?

File: common/src/main/java/com/revenuecat/purchases/common/Factories.kt [hunk 1]
```diff
@@ -124,6 +125,12 @@ private fun JSONObject.optPeriodType(name: String) = when (optString(name)) {
     else -> PeriodType.NORMAL
 }
 
+private fun JSONObject.getOwnershipType(name: String) = when (getString(name)) {
+    "PURCHASED" -> OwnershipType.PURCHASED
+    "FAMILY_SHARING" -> OwnershipType.FAMILY_SHARING
```
[the remark this answers is not in the document]
Comment (daentech) [reply common/src/main/java/com/revenuecat/purchases/common/Factories.kt]: Fixed

**Comment by daentech** [comment]
Fixed the FAMILY_SHARED key and added some tests

**Comment by aboedo** [comment]
Thanks for the updates! I'm taking a look at what's happening with tests. I don't think the issues are related to the PR, but I'm going to check whether there's a quick fix (other than suppressing warnings) as well as try to understand why they only started failing now, since the param list was already too long, like you said, before this PR

**Comment by daentech** [comment]
I wondered if the signing key was a secret and because it's a PR from a fork it won't be available

**Comment by daentech** [comment]
The lint is failing because I edited some files which already would have been failing and the check might only be on files edited in the current PR :)

**Comment by daentech** [comment]
I can fix the issue with the funciton limit by splitting the `Factories` file into a couple of smaller ones (Can probably break it down further), but I'm not sure about the parameter limit one

**Comment by daentech** [comment]
I've found a potential solution to the parameter limit lint failure. If I make `EntitlementInfo` a `data class` then it passes. I guess you can have data class constructors longer than normal methods

**Comment by aboedo** [comment]
We've slowly been moving towards data classes, so we should be fine. Thanks for the fixes! Approving and merging now 🚢 

(And yes, the remaining failing test is failing because we don't pass env variables to jobs in forks for security purposes).

**Review by aboedo — APPROVED** [review]

**Pull request merged**